src/OpenTelemetry/Batch.cs

```diff
@@ -45,10 +48,21 @@ internal Batch(CircularBuffer<T> circularBuffer, int maxSize)
             Debug.Assert(maxSize > 0, $"{nameof(maxSize)} should be a positive number.");
 
             this.item = null;
+            this.metrics = null;
             this.circularBuffer = circularBuffer ?? throw new ArgumentNullException(nameof(circularBuffer));
             this.targetCount = circularBuffer.RemovedCount + Math.Min(maxSize, circularBuffer.Count);
         }
 
+        internal Batch(T[] metrics, int maxSize)
+        {
+            Debug.Assert(maxSize > 0, $"{nameof(maxSize)} should be a positive number.");
```

Comment:
Do we still need some runtime check on `maxSize`, as `Debug.Assert` doesn't show up in release build?